### PR TITLE
Test canonical: reworked

### PR DESCRIPTION
**Pull request by RomanFedytskyi**

### CODE REVIEW

**Comment by zalesky** [comment]
I prefer element.querySelector('.pass') to element.getElementsByClassName('pass')[0]

File: tests/testcanonicalization.js [hunk 1]
```diff
@@ -0,0 +1,89 @@
+/**
+ * @fileoverview Defines test for Canonicalization widget.
+ * Successful criteria:
```
Comment (AlbaMontis) [inline tests/testcanonicalization.js]: "Success criteria"

Comment (RomanFedytskyi) [inline tests/testcanonicalization.js]: fixed

**Comment by AlbaMontis** [comment]
LGTM

File: tests/testcanonicalization.js [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+ * @fileoverview Defines test for Canonicalization widget.
+ * Success criteria:
+ * - Element with ID 'widget-canonical-url' should exist;
+ * - Element with ID 'widget-canonical-url' should contain elements with
+ *   ID 'canonical-url' and element with ID 'canonical-ip';
+ * - Element with ID 'canonical-url' and element with ID 'canonical-ip' should
+ *   contain element with CSS class 'pass' or element with CSS class 'fail';
+ * - Element with CSS class 'pass' should contain string 'Yes';
+ * - Element with CSS class 'fail' should contain string 'No';
+ * - Element with CSS class 'pass' should have CSS style 'display: block' if
```
Comment (mshchygol) [inline tests/testcanonicalization.js]: Suggestion:  
...should have CSS style property 'display: block'

File: tests/testcanonicalization.js [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+function testCanonicalization() {
+  // Element with ID 'widget-canonical-url' should exist.
+  /** @type {boolean} */
+  var result = !document.querySelector(WIDGET_SELECTOR);
+  /** @type {Array.<string>} */ var rows = [CANONICAL_URL_ID, CANONICAL_IP_ID];
+  /** @type {number} */ var length = rows.length;
+  /** @type {Element} */ var element;
+  /** @type {Element} */ var passElement;
+  /** @type {Element} */ var failElement;
+  /** @type {Element} */ var passDisplay;
```
Comment (mshchygol) [inline tests/testcanonicalization.js]: Type of this variable is not {Element}, getPropertyValue('display') returns {string}.

**Comment by mshchygol** [comment]
LGTM

File: tests/testcanonicalization.js [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+  /** @type {Array.<string>} */ var rows = [CANONICAL_URL_ID, CANONICAL_IP_ID];
+  /** @type {number} */ var length = rows.length;
+  /** @type {Element} */ var element;
+  /** @type {Element} */ var passElement;
+  /** @type {Element} */ var failElement;
+  /** @type {string} */ var passDisplay;
+  /** @type {string} */ var failDisplay;
+
+  if (!result) {
+    for (; length;) {
```
Comment (zalesky) [inline tests/testcanonicalization.js]: I prefer 
`for (; length--;) `

**Comment by zalesky** [comment]
LGTM

**Comment by vasylsemiliak** [comment]
LGTM

**Comment by Alexsv7** [comment]
LGTM

File: tests/testcanonicalization.js [hunk 5]
```diff
@@ -0,0 +1,89 @@
+/**
+ * @fileoverview Defines test for Canonicalization widget.
+ * Success criteria:
+ * - Element with ID 'widget-canonical-url' should exist;
+ * - Element with ID 'widget-canonical-url' should contain elements with
+ *   ID 'canonical-url' and element with ID 'canonical-ip';
+ * - Element with ID 'canonical-url' and element with ID 'canonical-ip' should
+ *   contain element with CSS class 'pass' or element with CSS class 'fail';
+ * - Element with CSS class 'pass' should contain string 'Yes';
```
Comment (vpodk) [inline tests/testcanonicalization.js]: Please do not hardcode localized values.

> ... content should not be empty, null or undefined.

**Pull request merged**